packages/components/transfer/src/composables/use-computed-data.ts

```diff
@@ -0,0 +1,42 @@
+import { computed } from 'vue'
+import { usePropsAlias } from './use-props-alias'
+
+import type { TransferDataItem, TransferKey, TransferProps } from '../transfer'
+
+export const useComputedData = (props: TransferProps) => {
+  const propsAlias = usePropsAlias(computed(() => props.props))
+
+  const dataObj = computed(() =>
+    props.data.reduce((o, cur) => (o[cur[propsAlias.value.key]] = cur) && o, {})
+  )
+
+  const sourceData = computed(() =>
+    props.data.filter(
+      (item) => !props.modelValue.includes(item[propsAlias.value.key])
+    )
+  )
```

Comment:
Can it be modified like this?

```suggestion
  const sourceData = computed(() =>
    props.data.filter((item) => {
      return (
        keysOf(item).includes(propsAlias.value.key) &&
        !props.modelValue.includes(item[propsAlias.value.key])
      )
    })
  )
```

Reply:
Sometimes there are problems with incoming data, for example, some labels or keys are null or undefined. We should try to avoid component crashes.

Check out the suggestions below